packages/m2m-gateway/test/api/purposes/getPurpose.test.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it, expect, vi } from "vitest";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole, genericLogger } from "pagopa-interop-commons";
+import request from "supertest";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { generateId } from "pagopa-interop-models";
+import { api, mockPurposeService } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import { getMockedApiPurpose } from "../../mockUtils.js";
+import { toM2MGatewayApiPurpose } from "../../../src/api/purposeApiConverter.js";
+import {
+  purposeNotFound,
+  missingActivePurposeVersion,
+} from "../../../src/model/errors.js";
+
+describe("GET /purpose/:purposeId router test", () => {
+  const authorizedRoles: AuthRole[] = [
+    authRole.M2M_ADMIN_ROLE,
+    authRole.M2M_ROLE,
+  ];
+
+  const makeRequest = async (token: string, purposeId: string) =>
+    request(api)
+      .get(`${appBasePath}/purposes/${purposeId}`)
+      .set("Authorization", `Bearer ${token}`)
+      .send();
+
+  const mockApiPurpose = getMockedApiPurpose();
+
+  const mockM2MPurposesResponse: m2mGatewayApi.Purpose = toM2MGatewayApiPurpose(
+    {
+      purpose: mockApiPurpose.data,
+      logger: genericLogger,
+    }
+  );
+
+  it.each(authorizedRoles)(
+    "Should return 200 and perform API clients calls for user with role %s",
```

Comment:
My bad for this typo in the first test setup, API tests don't test API clients calls

```suggestion
    "Should return 200 and perform service calls for user with role %s",
```